liquibase-standard/src/main/java/liquibase/diff/output/changelog/DiffToChangeLog.java

```diff
@@ -493,6 +503,10 @@ private List<DatabaseObject> sortObjects(final String type, Collection<DatabaseO
                 }
             } catch (DatabaseException e) {
                 Scope.getCurrentScope().getLog(getClass()).fine("Cannot get object dependencies: " + e.getMessage());
+            } catch (StackOverflowError e) {
```

Comment:
I don't like this, but I'm not able to reproduce it so I'm treating the symptoms without addressing the underlying cause.